### ISSUE TITLE
Question: GCD and multi-threading

### ISSUE THREAD

**Issue by RenniePet** [IBM-Swift/BlueSocket#40]
Hope you have a moment to answer a newbie question - am completely new to Mac and iOS and Swift.

Was glad to find this project. Have looked at several other Swift projects for TCP but they're obsolete (written in older versions of Swift) or mix Objective-C with Swift or are overkill for my expected usage. Have now succeeded in building this project and running the test echo server program on a Mac.

My question: Am I correct in understanding that BlueSocket itself does not provide any multi-threading services at all - it is simply a wrapper for the Socket API in the operating system? And that GCD is used to provide multi-threading in the test echo server program?

Thanks.

**Comment by billabt** [comment]
Yes, you're correct.  BlueSocket does not provide any multi-threading services.  **Dispatch** (aka GCD) is providing the multi-threading capability in test program.  **Dispatch** is bundled with and part of the Swift toolchain.
